src/slurm_plugin/instance_manager.py

```diff
@@ -358,3 +361,90 @@ def terminate_all_compute_nodes(self, terminate_batch_size):
     def _update_failed_nodes(self, nodeset, error_code="Exception"):
         """Update failed nodes dict with error code as key and nodeset value."""
         self.failed_nodes[error_code] = self.failed_nodes.get(error_code, set()).union(nodeset)
+
+    def get_console_output_task(self, nodes, output_callback, wait_times):
```

Comment:
The code that retrieves the console output has been moved. However, I think it makes sense to leave the code that does the node -> instance id mapping in instance manager since instance manager is the owner of that data.